ensemble/volren/volume_data.py

```diff
@@ -75,6 +76,10 @@ class VolumeData(HasStrictTraits):
     # The data as a fortran array
     _raw_data = VolumeArray
 
+    # The filter to apply to the raw data
+    volume_filter = Property(Instance(VolumeFilter))
+    _volume_filter = Instance(VolumeFilter)
```

Comment:
Good point, please have `volume_filter` be an `Instance(VolumeFilter)`, and add a `_volume_filter_changed` listener that updates `_render_data`. Note that you don't need to re-sample and re-mask the data every time you change the filter, it's best to keep the mask data around.